src/client/game.cpp

```diff
@@ -1127,7 +1133,9 @@ bool Game::startup(bool *kill,
 	device                    = m_rendering_engine->get_raw_device();
 	this->kill                = kill;
 	this->error_message       = &error_message;
-	reconnect_requested       = reconnect;
+	this->reconnect_requested = reconnect;
+	this->reconnect_address   = &reconnect_address;
```

Comment:
This seems fragile in terms of lifetimes. What if the strings given to us don't live long enough? Wouldn't it make more sense to take ownership if them?